### ISSUE TITLE
Max Call Stack exceeded for slip collisions

### ISSUE THREAD

**Issue by brandonLi8** [phetsims/collision-lab#129]
With the last preset for the 'Inelastic' screen on "slip", there appears to be infinite recursion. Strangely, the problem goes away when the reset button is pressed. Looking into it.

**Comment by brandonLi8** [comment]
Able to reproduce. Doesn't seem to happen when the step button is pressed with `?timeStepDuration=1`

**Comment by brandonLi8** [comment]
Fixed in the commits above. Closing

**Comment by brandonLi8** [comment]
Reopening, the changes that I implemented above works for this scenario. However, now multiple balls colliding at the same time is not supported.

**Comment by brandonLi8** [comment]
I committed a workaround above. Essentially, I've figured out that the problem is that after the collision occurs, Ball2's velocity is tangential to Ball1's velocity (instead of "away" from) and the balls are tangentially touching.
![image](https://user-images.githubusercontent.com/42391580/89066994-ab58cf00-d32b-11ea-83a7-e613536b000e.png)

Our collision detection then thinks that the collision still is happening on each step call, causing a max call stack error. This problem is, thus, impossible for all other screen, since there are not perfectly inelastic collisions for Explore 2D and the other screens are 1D.

The workaround is I implemented was to step the balls by 1E-14 seconds after a slip inelastic collision occurs in 2D. This will move the balls forwards by an small quantity (around 1E-14), meaning the balls are not tangentially touching and collisions are no longer detected.
